sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -173,22 +173,38 @@ private ExecutorRequirement getExecutorRequirement(
             String serviceName,
             UUID targetConfigurationId) throws InvalidRequirementException {
         List<Protos.TaskInfo> podTasks = TaskUtils.getPodTasks(podInstance, stateStore);
+        Protos.ExecutorInfo executorInfo = null;
 
         for (Protos.TaskInfo taskInfo : podTasks) {
             Optional<Protos.TaskStatus> taskStatusOptional = stateStore.fetchStatus(taskInfo.getName());
             if (taskStatusOptional.isPresent()
                     && taskStatusOptional.get().getState() == Protos.TaskState.TASK_RUNNING) {
+                executorInfo = taskInfo.getExecutor();
                 LOGGER.info(
                         "Reusing executor from task '{}': {}",
                         taskInfo.getName(),
-                        TextFormat.shortDebugString(taskInfo.getExecutor()));
-                return ExecutorRequirement.create(taskInfo.getExecutor());
+                        TextFormat.shortDebugString(executorInfo));
             }
         }
 
+        Map<String, Protos.Resource> volumeMap = new HashMap<>();
+        if (executorInfo == null) {
```

Comment:
The logic here is basically the same as in [DefaultOfferRequirementProvider#149](https://github.com/mesosphere/dcos-commons/blob/824ee8146ecf56fbf50fb364c2268a3affa0617e/sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java#L149): to do the right bookkeeping for resources on an "existing" requirement, we need to pass the existing `Resource` protobuf around so that the resource ID ends up on the `TaskInfo` that we launch. I suppose, thinking about this a little more, that that's unnecessary -- executor resources can't change after launch (and offers aren't evaluated for executor resources in the case that we're placing a task on an already-running executor), so we may as well not even bother.